### DISCUSSION TITLE
Raspberry Pi 5 + RPi OS Bookworm

### DISCUSSION THREAD

**Discussion by SvenVD** [OpenSurv/OpenSurv#21] [Hardware + Operating system]
All experiences and tests with OpenSurv on Raspberry Pi 5 and RPi OS Bookworm

For Raspberry Pi 5 + Ubuntu 24.04 LTS look here  https://github.com/OpenSurv/OpenSurv/discussions/3

**Comment by SvenVD** [comment]
Old thread https://github.com/OpenSurv/OpenSurv/discussions/3#discussioncomment-11532417

**Comment by aniceberg** [comment]
I noticed that after a varying number of hours, that the video feeds can start to have a delay of anywhere from 5-10 seconds. I added a daily reboot to my crontab to help alleviate issues but I'm thinking to do it hourly. Or at least an hourly restart of the lightdm service.

**Comment by skowron0** [comment]
Hello, in an earlier post I wrote that on Raspberry Pi 5 8GB with RPi OS Bookworm the image from the cameras worked but after turning on the power it did not start by itself, I had to manually enter the restart command. I do not know why this happened but after a few attempts I formatted the card again and uploaded the latest version of RPi OS (Recommended with desktop environment 64bit) and reinstalled OpenSurv. Surprisingly, now everything works perfectly - I probably made a mistake earlier. I have been testing the playback of 4 H265 streams with a resolution of 720p for the second day and they work perfectly. Thank you very much for a great tool!

**Reply by ByWaleed** [reply]
In case you or anyone else are in a situation where you don't seen any video output after turning the device on, try running `systemctl restart lightdm.service` remotely via ssh. If such issue persists, you can set a cron job to run the aforementioned command on a schedule (not ideal but could be effective).

**Comment by turbo2ltr** [comment]
If this thread doesn't apply to the lite version, feel free to move.

Pi 5, 4GB, brand new.  Fresh install of RPi OS Lite 64bit (bookworm).
Installed Opensurv.
The first time I installed it, /etc/opensurv was empty.
```
rpisurv pi [/etc/opensurv] # ll
total 8.0K
drwxr-xr-x 2 opensurv opensurv 4.0K Apr  6 21:01 .
drwxr-xr-x 8 opensurv opensurv 4.0K Apr  6 21:08 ..
rpisurv pi [/etc/opensurv] #
```
I looked through the install output and didn't see any errors.

I tried to start opensurv with `systemctl restart lightdm.service` (via SSH) and very oddly, I got a desktop login screen.  I thought the lite version didn't have the desktop.

So I ran the install again.  And this time I had a symlink in `/etc/opensurv`
Didn't really help. If I start lightdm.service, then I get a desktop and I can log in if I want.  If I stop the service the desktop goes away.
Not sure how I screwed this up.

**Reply by turbo2ltr** [reply]
I tried again just in case. This time I did it on the actual console instead of over SSH.
Complete fresh OS image. Booted and immediately did the following:
```
sudo apt-get update
sudo apt install git
git clone https://github.com/OpenSurv/OpenSurv
cd OpenSurv
sudo ./install.sh
```
It installed something like 400 packages ~250mb.
I said yes to overwriting the existing config with the default though there couldn't be any existing as it was a brand new os image.
Then it said it needed to reboot 
When it came back up it just presented me with the usual console login.
I logged in.
I typed `systemctl restart lightdm.service` and it gave me an authentication message with red text which is something I've personally never seen before but I also haven't used bookworm.
```
pi@rpisurv:~ $ systemctl restart lightdm.service
==== AUTHENTICATING FOR org.freedesktop.systemd1.manage-units ====
Authentication is required to restart 'lightdm.service'.
Authenticating as: ,,, (pi)
Password:
```
In any case I assume that's normal. running it with sudo seems to work without the prompt.
The issue is it worked as in it just launches the desktop login screen.  So I don't think it's me.  Must be a compatibility issue with the Lite version of the OS..

**Reply by turbo2ltr** [reply]
I figured it out!  It's missing the yaml python package.

`sudo apt install python3-yaml` and then `sudo systemctl restart lightdm.service` and it's working!  Now to see the performance.

**Reply by turbo2ltr** [reply]
Running 7 ubiquiti cameras on a 4K monitor (mix of medium and low streams) and appears to be good.  Load varies between 1.8 and 2.3.  I'm using the waveshare POE hat that has a very quiet fan and temps appear to be holding at 50c. Probably close to 26C in the room right now. 2.1gb free memory.  WAY less heat that the HP EliteDesk 800G3 I also tested. According to the Unifi switch, it's drawing 6.76W POE.

**Reply by turbo2ltr** [reply]
One other issue, was booting into the CLI even though lightdm.service was set to start on boot. Needed:
`sudo systemctl set-default graphical.target`

**Reply by SvenVD** [reply]
Thanks for reporting back and happy you were able to make it work!

**Reply by rehajel** [reply]
Thanks for the thread!
With this information I managed to get opensurv running on a pi zero2w on raspios lite with one 720p stream. Still testing at the moment, but it seems to mostly hold up if I adjust the init_stream timeout to 15 and the probe timeout to 10. Would not recommend this setup though.

**Reply by ajquick** [reply]
> I figured it out! It's missing the yaml python package.
> 
> `sudo apt install python3-yaml` and then `sudo systemctl restart lightdm.service` and it's working! Now to see the performance.

Thank you for it. I had to do it with my Raspberry as well.

**Comment by linek1980** [comment]
I have a huge increase performance in PI5 : OS Raspberry Pi OS, Debian Trixie. 
freeform_advanced_mpv_options: "--rtsp-transport=udp --profile=low-latency --untimed --opengl-glfinish=yes --gpu-dumb-mode=no --hwdec=drm-copy --vo=gpu"

If your camera has lag, use additional --vd-lavc-threads=1  if the delay is still there change vo=gpu to --vo=sdl

**Reply by turbo2ltr** [reply]
Interesting. I tried these and all the cameras would freeze after about 4 seconds. Not all of them even loaded. Though I'm still on bookworm not trixie..

**Comment by linek1980** [comment]
did you know that you have to use rules for each stream ? 

example

 - url: "rtsp://192.168.1.11:7447/avOdAnMTMfkeV76Z"
          freeform_advanced_mpv_options: "--rtsp-transport=udp --profile=low-latency --untimed --opengl-glfinish=yes --gpu-dumb-mode=no --hwdec=drm-copy --vo=gpu"
        - url: "rtsp://192.168.1.11:7447/PtKP5DXyVP8hOuCa"
          freeform_advanced_mpv_options: "--rtsp-transport=udp --profile=low-latency --untimed --opengl-glfinish=yes --gpu-dumb-mode=no --hwdec=drm-copy --vo=gpu"
